### DISCUSSION TITLE
Remove background from borders / Add borders to preview

### DISCUSSION THREAD

**Discussion by Zykatious** [saghen/blink.cmp#620] [Q&A] [answered]
Hey all, have been trying to configure blink to have rounded borders to be consistent with my existing setup, however when I enable them they have the background colour bleeding out beyond the border.
In nvim-cmp that could be got around by having vim.opt.pumblend = 0, but it's not doing the same here.
Also, I can't seem to figure out how to put borders around the preview of the selected item. 
Any help would be appreciated!

Have attached a screenshot to show what I mean
<img width="1009" alt="Screenshot 2024-12-17 at 15 29 59" src="https://github.com/user-attachments/assets/d8d7c172-97b9-42ce-babe-0c621c4af3d4" />

**Comment by soifou** [comment]
To avoid "bleeding" borders, I usually use the following box drawings `{ '🭽', '▔', '🭾', '▕', '🭿', '▁', '🭼', '▏' }`. That's the only one I'm aware of that have a clean look for popups (and floating windows). You'll loose your round corners though.

As for your border problem and by looking at your screenshot, your `documentation` block must be nested in `completion`:

```lua
completion = {
    documentation = {
        window = {
            border = 'rounded',
        },
    },
},
```

**Comment by idr4n** [comment, marked as answer]
I have done with this:

```lua
      completion = {
        list = {
          selection = "auto_insert",
        },
        documentation = {
          auto_show = true,
          auto_show_delay_ms = 200,
          window = {
            border = "rounded",
            winhighlight = "Normal:Normal,FloatBorder:FloatBorder,CursorLine:BlinkCmpDocCursorLine,Search:None",
          },
        },
        menu = {
          border = "rounded",
          draw = { gap = 2 },
          winhighlight = "Normal:Normal,FloatBorder:FloatBorder,CursorLine:BlinkCmpMenuSelection,Search:None",
        },
```

Change `FloatBorder:FloatBorder` to `FloatBorder:<Your Highlight Group>`. Note that the default `winhighlight` is:

```lua
-- for menu
winhighlight = 'Normal:BlinkCmpMenu,FloatBorder:BlinkCmpMenuBorder,CursorLine:BlinkCmpMenuSelection,Search:None',
--- for docs
winhighlight = 'Normal:BlinkCmpDoc,FloatBorder:BlinkCmpDocBorder,CursorLine:BlinkCmpDocCursorLine,Search:None',
```

**Reply by Zykatious** [reply]
This is perfect, thank you!

**Reply by ntrkd** [reply]
Hint for anyone before you spend 20 minutes debugging - the first code block (menu section) contains a space after ```':'``` in ```CursorLine: BlinkCmpMenuSelection```.

Be sure to strip that space or else it will throw a ```E5248: Invalid character in group name```

**Comment by anuragkumar2921** [comment]
worked like charm

**Comment by DMunkei** [comment]
Hey, I don't quiet get what understand how to get this working...? :D I'm using Kanagawa. What do you mean reference your highlight group?

**Reply by idr4n** [reply]
Hi,
It means that you need to choose the highlight group from your colorscheme that suits your needs in terms of the foreground and background color.

Alternatively, I think you can modify the Blink highlight groups in Kanagawa settings, for example (I haven't tried this though as I don't use Kanagawa):

```lua
    overrides = function(colors) -- add/modify highlights
      local palette = colors.palette
      return {
        BlinkCmpMenu = { bg = "none" },
        BlinkCmpDoc = { bg = "none" },
        BlinkCmpMenuBorder = { fg = palette.oniViolet },
        BlinkCmpDocBorder = { fg = palette.oniViolet },
      }
    end,
```

If you do this, you dont need to add these lines as mentioned in answer above

```lua
winhighlight = "Normal:Normal,FloatBorder:FloatBorder,CursorLine:BlinkCmpDocCursorLine,Search:None",

winhighlight = "Normal:Normal,FloatBorder:FloatBorder,CursorLine:BlinkCmpMenuSelection,Search:None",
```
